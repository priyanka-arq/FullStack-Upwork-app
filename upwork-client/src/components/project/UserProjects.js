import React, { Component } from "react";
import * as projectActions from "../../redux/actions/projectActions";
import { connect } from "react-redux";
import UserProjectsDetails from "./UserProjectsDetails";
import jobImage from "../Images/jobImage.png";
import styled from "styled-components";

const ProjectContainer = styled.div`
  width: 100;
  height: 100;
  justify-content: center;
`;

export class UserProjects extends Component {
  constructor(props) {
    super(props);
    this.state = {
      projects: [],
      redirect: "",
      projectId: props.match.params.projectId,
    };
  }
  componentDidMount() {
    const userProjects = this.props.projects;

    if (userProjects.length === 0) {
      this.props.loadProjects().catch((error) => {
        alert("Failed to load Profile " + error);
      });
    }

    const projects = userProjects.filter(
      (project) => project.user === this.props.match.params.userId
    );
    this.setState({ projects: projects });
    console.log();
  }
  handleClick = () => {
    this.props.history.push(`/project`);
  };
  render() {
    return (
      <div>
        {this.state.project ? (
          <UserProjectsDetails
            projects={this.state.projects}
            handleUpdate={this.handleUpdate}
          />
        ) : (
          <div>
            <h2>Post your job here? </h2>
            <button className="button" onClick={this.handleClick}>
              Post your job
            </button>
            <ProjectContainer>
              <img id="phoneImage" src={jobImage} />
            </ProjectContainer>
          </div>
        )}
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => ({
  projects: state.projects,
});

const mapDispatchToProps = (dispatch) => ({
  loadProjects: () => dispatch(projectActions.loadProjects()),
});

export default connect(mapStateToProps, mapDispatchToProps)(UserProjects);
