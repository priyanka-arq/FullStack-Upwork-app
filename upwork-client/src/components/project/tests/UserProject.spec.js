import React from "react";
import { shallow } from "enzyme";
import { UserProjects } from "../UserProjects";
import projects from "../../../fixtures/projects";

let wrapped, handleClick, history, loadProjects;

beforeEach(() => {
  handleClick = jest.fn();
  loadProjects = jest.fn();
  history = { push: jest.fn() };
  wrapped = shallow(
    <UserProjects
      match={{ params: { projectId: 1 } }}
      projects={projects}
      history={history}
      onClick={handleClick}
      loadProjects={loadProjects}
    />
  );
});

afterEach(() => {
  wrapped.unmount();
});

test("Should render UserProjects correctly ", () => {
  expect(wrapped).toMatchSnapshot();
});

it("Should render button correctly ", () => {
  expect(wrapped.find("button").length).toBe(1);
});

test("Should handle on click event on button", () => {
  wrapped.find(".button").simulate("click");
  expect(history.push).toHaveBeenLastCalledWith("/project");
});
