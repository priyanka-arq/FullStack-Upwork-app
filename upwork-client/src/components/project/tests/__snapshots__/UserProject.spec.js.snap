// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render UserProjects correctly  1`] = `
<div>
  <div>
    <h2>
      Post your job here? 
    </h2>
    <button
      className="button"
      onClick={[Function]}
    >
      Post your job
    </button>
    <styled.div>
      <img
        id="phoneImage"
        src="jobImage.png"
      />
    </styled.div>
  </div>
</div>
`;
